Modules/modFileEncoding.ACM

```diff
@@ -314,6 +284,85 @@ Function UTF8_to_UCS2(bytsUTF8() As Byte, ByRef bytsUCS2() As Byte) As Boolean
+  ' We store these as longs (32 bit) and each value is a maximum of 14 bits so we don't need to worry about any issues with MS Access not having unsigned ints
+  
+  If Not lngsMap Then
+    ' The array has not been populated yet
+    ReDim lngsMap(128 To 159)
+    ' unused codes &H81 (129), &H8D (141), &H8F (143), &H90 (144) and &H9D (157) are mapped to '?' &H003F
+    lngsMap(128) = &H20AC: lngsMap(129) = &H3F:   lngsMap(130) = &H201A: lngsMap(131) = &H192
+    lngsMap(132) = &H201E: lngsMap(133) = &H2026: lngsMap(134) = &H2020: lngsMap(135) = &H2021
+    lngsMap(136) = &H2C6:  lngsMap(137) = &H2030: lngsMap(138) = &H160:  lngsMap(139) = &H2039
+    lngsMap(140) = &H152:  lngsMap(141) = &H3F:   lngsMap(142) = &H17D:  lngsMap(143) = &H3F
+    lngsMap(144) = &H3F:   lngsMap(145) = &H2018: lngsMap(146) = &H2019: lngsMap(147) = &H201C
+    lngsMap(148) = &H201D: lngsMap(149) = &H2022: lngsMap(150) = &H2013: lngsMap(151) = &H2014
+    lngsMap(152) = &H2DC:  lngsMap(153) = &H2122: lngsMap(154) = &H161:  lngsMap(155) = &H203A
+    lngsMap(156) = &H153:  lngsMap(157) = &H3F:   lngsMap(158) = &H17E:  lngsMap(159) = &H178
+  End If
+  
+End Function
+
+Function ANSICP_to_UnicodeCP(bytANSI As Byte, ByRef bytUnicodeMSB As Byte, ByRef bytUnicodeLSB As Byte) As Boolean
+  ' Takes a single byte ANSI codepoint (character) and converts it to 16 bit Unicode placing the answer
+  ' in 2 separate Bytes representing the MSB (Most Significant Byte) and LSB (Least Significant Byte)
+  Static bytsUnicodeBestFit() As Long
+  Const lngLSBMask = 255 ' constant with Least Significant 8 bits set to 1
+  
+  ANSICP_to_UnicodeCP = False ' initially set to False
+  
+  If bytANSI < 128 Or bytANSI >= 160 Then
+    bytUnicodeMSB = 0
+    bytUnicodeLSB = bytANSI
+  Else
+    ' in this range we neede to use the best  fit lookup table
+    Call PopulateANSI_Unicode_BestFit(bytsUnicodeBestFit)
+    bytUnicodeMSB = Fix(bytsUnicodeBestFit(bytANSI) / 256)
+    bytUnicodeLSB = bytsUnicodeBestFit(bytANSI) And lngLSBMask
+  End If
+        
+  ANSICP_to_UnicodeCP = True
+  
+End Function
+
+Function UnicodeCP_to_ANSICP(bytUnicodeMSB As Byte, bytUnicodeLSB As Byte, ByRef bytANSI As Byte) As Boolean
+  ' Take a 2 byte ANSI codepoint and converts it to 8 bit ANSI placing the answer in the byt provided
+  ' MSB = Most Significant Byte, LSB = Least Significant Byte
+  ' Returns False if the character cannot be mapped
+  
+  Static bytsUnicodeBestFit() As Long
+  Dim lngUnicode As Long
+  Dim i As Long
+  
+  UnicodeCP_to_ANSICP = False ' intially set to False
+  
+  If bytUnicodeMSB = 0 Then
+    ' we have a single byte value
+    If bytUnicodeLSB < 128 Then
+      bytANSI = bytUnicodeLSB
+      UnicodeCP_to_ANSICP = True
+    ElseIf bytUnicodeLSB >= 160 Then
+      bytANSI = lngUnicode
+      UnicodeCP_to_ANSICP = True
+    'Else
+      ' we have a value in the range 128 (&H80) to 9F (159) which shouldn't be possible so do nothing to return a fail

```

Comment:
Maybe explicitly `Exit Function` here? It is a little confusing seeing a commented else.

Also there are a mix of decimal and hex
```suggestion
      ' we have a value in the range 80 (128) to 9F (159) which shouldn't be possible so do nothing to return a fail
```

Reply:
Done